IllegalAccessProtectedMethodTest_hs_0 fails on dynamic node due to missing jcommander.jar

Failure link
------------

Public link to the failing test: https://ci.adoptium.net/job/Grinder/17178/
[Grinder Link](https://ci.adoptium.net/job/Grinder/parambuild/?SDK_RESOURCE=upstream&amp;TARGET=testList+TESTLIST%3DIllegalAccessProtectedMethodTest_hs_0&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=5&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_mac&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;RERUN_ITERATIONS=0&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=&amp;BUILD_IDENTIFIER=&amp;ADOPTOPENJDK_BRANCH=v1.0.13-release&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;NON_AQA_TEST_REPOS=&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=&amp;JDK_VERSION=8&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Fjdk8u&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Frelease%2Fjobs%2Fjdk8u%2Fjdk8u-release-mac-x64-temurin&amp;CLOUD_PROVIDER=&amp;PLATFORM_AND_MACHINE=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=jdk8u492-b09_adopt&amp;TEST_IMAGES_REQUIRED=true&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=hotspot&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=aqavit-bot&amp;DYNAMIC_COMPILE=true&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=eclipse_temurin_bot_email_and_token&amp;ARCHIVE_TEST_RESULTS=false&amp;OPENJDK_SHA=&amp;NUM_MACHINES=&amp;RERUN_FAILURE=false&amp;BUILD_LIST=functional&amp;USE_TESTENV_PROPERTIES=true&amp;UPSTREAM_JOB_NUMBER=46&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=25&amp;JVM_OPTIONS=&amp;PARALLEL=None)

```
10:06:09  Error: A JNI error has occurred, please check your installation and try again
10:06:09  Exception in thread "main" java.lang.NoClassDefFoundError: com/beust/jcommander/ParameterException
10:06:09  	at java.lang.Class.getDeclaredMethods0(Native Method)
10:06:09  	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
10:06:09  	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
10:06:09  	at java.lang.Class.getMethod0(Class.java:3018)
10:06:09  	at java.lang.Class.getMethod(Class.java:1784)
10:06:09  	at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:670)
10:06:09  	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:652)
10:06:09  Caused by: java.lang.ClassNotFoundException: com.beust.jcommander.ParameterException
10:06:09  	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
10:06:09  	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
10:06:09  	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
10:06:09  	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
10:06:09  	... 7 more
```

The problem is because this is running on a github dynamic runner agent, it has no existing "externalDependency" LIB cache, and because the testcase does not have "jcommander"in the dependency LIB list, it does not download it, which it should. https://github.com/eclipse-openj9/openj9/blob/57063095cb1c53bafe6279733449262913dec54c/test/functional/IllegalAccessError_for_protected_method/build.xml#L38

--- a/test/functional/IllegalAccessError_for_protected_method/build.xml
+++ b/test/functional/IllegalAccessError_for_protected_method/build.xml
@@ -35,7 +35,7 @@
 	<property name="src" location="./src" />
 	<property name="build" location="./bin" />
 	<property name="transformerListener" location="${TEST_ROOT}/Utils/src"/>
-	<property name="LIB" value="asm_all,testng"/>
+	<property name="LIB" value="asm_all,testng,jcommander"/>
 	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml"/>
 
 	<target name="init">
@@ -54,6 +54,7 @@
 			<src path="${src}" />
 			<classpath>
 				<pathelement location="${LIB_DIR}/asm-all.jar" />
+				<pathelement location="${LIB_DIR}/jcommander.jar" />
 				<pathelement location="${LIB_DIR}/testng.jar" />
 			</classpath>
 		</javac>

